File: BST/Search/sol.py
# BST Node
class Node:
    def __init__(self, val):
        self.data = val
        self.left = None
        self.right = None


class Solution:
    def search(self, root, key):
        while root:
            if root.data == key:
                return True
            elif root.data > key:
                root = root.left
            else:
                root = root.right

        return False

File: BST/Search/sol.cpp
#include <iostream>
using namespace std;

class Node {
public:
    int data;
    Node *left;
    Node *right;

    Node(int val) {
        data = val;
        left = right = nullptr;
    }
};


class Solution {
  public:
    bool search(Node* root, int key) {
        // code here
        while(root){
            if(root->data==key)return true;
            else if(root->data > key)root = root->left;
            else root = root->right;
        }
        return false;
    }
};